### PR TITLE
Use -fno-common for sim KERNEL build

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

File: boards/sim/sim/sim/scripts/Make.defs [hunk 1]
```diff
@@ -63,6 +63,11 @@ ARCHINCLUDES = -I. -isystem $(TOPDIR)/include
 ARCHINCLUDESXX = -I. -isystem $(TOPDIR)/include -isystem $(TOPDIR)/include/cxx
 ARCHSCRIPT =
 
+ifneq ($(KERNEL),y)
```
Comment (xiaoxiang781216) [inline boards/sim/sim/sim/scripts/Make.defs]: Why skip -fno-common for KERNEL build?

File: boards/sim/sim/sim/scripts/Make.defs [hunk 2]
```diff
@@ -63,6 +63,11 @@ ARCHINCLUDES = -I. -isystem $(TOPDIR)/include
...
+  ARCHCPUFLAGS += -fno-common
+  ARCHCPUFLAGSXX += -fno-common
+endif
```
Comment (xiaoxiang781216) [inline boards/sim/sim/sim/scripts/Make.defs]: need add to other Mak.defs under boards/sim/sim/sim/configs

File: boards/sim/sim/sim/scripts/Make.defs (hunk 1 above)
Comment (yamt) [reply boards/sim/sim/sim/scripts/Make.defs]: maybe a mistake. i'll check.

File: boards/sim/sim/sim/scripts/Make.defs (hunk 2 above)
Comment (yamt) [reply boards/sim/sim/sim/scripts/Make.defs]: probably.
the configs involving other make.defs don't build on macOS for unrelated reasons though.

Comment (xiaoxiang781216) [reply boards/sim/sim/sim/scripts/Make.defs]: But other people may try to build that config, it's better to fix all together.

Comment (yamt) [reply boards/sim/sim/sim/scripts/Make.defs]: ok

File: boards/sim/sim/sim/scripts/Make.defs (hunk 1 above)
Comment (yamt) [reply boards/sim/sim/sim/scripts/Make.defs]: i don't remember why i did the ifneq. i removed the conditional as it's necessary regardless of KERNEL.

**Pull request merged**